src/modules/85-cv/components/ChangeTimeline/ChangeTimeline.tsx

```diff
@@ -81,6 +99,35 @@ export default function ChangeTimeline(props: ChangeTimelineProps): JSX.Element
     environmentIdentifier
   ])
 
+  useEffect(() => {
+    MonitoredServiceChangeTimelineCancel()
+    MonitoredServiceChangeTimelineRefetch({
+      queryParams: {
+        accountId,
+        orgIdentifier,
+        projectIdentifier,
+        environmentIdentifier: Array.isArray(environmentIdentifier) ? environmentIdentifier[0] : environmentIdentifier,
+        serviceIdentifier: Array.isArray(serviceIdentifier) ? serviceIdentifier[0] : serviceIdentifier,
+        changeSourceTypes: changeSourceTypes || [],
+        duration: duration?.value as TimePeriodEnum,
+        endTime: Date.now()
+      }
+    })
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [duration])
```

Comment:
Worth adding the deps here
```suggestion
  }, [accountId, orgIdentifier, projectIdentifier, environmentIdentifier, serviceIdentifier, changeSourceTypes, duration])
```